src/downloader.cpp

```diff
@@ -180,23 +180,20 @@ Download* Downloader::startDownload(const std::string& uri, const std::vector<st
 Download* Downloader::getDownload(const std::string& did)
 {
   try {
-    m_knownDownloads.at(did).get()->updateStatus(true);
     return m_knownDownloads.at(did).get();
   } catch(std::exception& e) {
     for (auto gid : mp_aria->tellActive()) {
       if (gid == did) {
         m_knownDownloads[gid] = std::unique_ptr<Download>(new Download(mp_aria, gid));
-        m_knownDownloads.at(gid).get()->updateStatus(true);
         return m_knownDownloads[gid].get();
       }
     }
     for (auto gid : mp_aria->tellWaiting()) {
       if (gid == did) {
         m_knownDownloads[gid] = std::unique_ptr<Download>(new Download(mp_aria, gid));
-        m_knownDownloads.at(gid).get()->updateStatus(true);
         return m_knownDownloads[gid].get();
       }
-  }
+    }
```

Comment:
A download may be missed in unlucky scenarios when it moves from the waiting list to the active list after the call to `mp_aria->tellActive()` has completed but before the first loop is done and `mp_aria->tellWaiting()` is called. A similar problem is present in the constructor of `Downloader`.